app/__main__.py

```diff
@@ -0,0 +1,107 @@
+import abc
+import csv
+import datetime
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union, Dict, List, Optional, Generator
+
+import dateutil
+import dateutil.parser
+
+
+@dataclass
+class Person:
+    address: str
+    data: Dict[str, Union[str, datetime.datetime]]
+
+
+class BaseResource(abc.ABC):
+
+    @abc.abstractmethod
+    def connect(self):
+        ...
+
+    @abc.abstractmethod
+    def parse(self):
+        ...
+
+
+class CSVResource(BaseResource):
+    def __init__(self, path):
+        self._path = path
+        self._reader: Optional[List[Dict[str, Union[str, datetime]]]] = None
+
+    def connect(self) -> None:
+        with open(self._path, mode='r') as csv_input:
+            self._reader = list(csv.DictReader(csv_input))
+
+    def parse(self) -> Generator:
+        assert self._reader
+        for person in self._reader:
+            yield Person(address=person[' email'], data=person)
+
+
+class Handler:
+    def __init__(self, resource: BaseResource):
+        self._resource = resource
+
+    def _select_birthday_person(self, person_data):
+        person_birthday = dateutil.parser.parse(person_data[" date_of_birth"])
+
+        if int(person_birthday.month) == datetime.datetime.today().month == 2 \
+                and person_birthday.day == '29' and datetime.datetime.today().day == 28:
+            return True
+
+        elif person_birthday.day == datetime.date.today().day \
+                and int(person_birthday.month) == datetime.datetime.today().month:
+            return True
+
+        return False
+
+    def build_birthdays_people(self):
+        persons = self._resource.parse()
+        persons_with_today_birthday = filter(lambda person: self._select_birthday_person(person.data), persons)
+        return persons_with_today_birthday
+
+
+class Sender:
+    def __init__(self, handler, service, message='Happy Birthday from Coding Dojo 2020!'):
```

Comment:
There can be multiple scenarios involved regarding the message.
Lets say that the Client wants to specify the message but as a template.
Example:
```
Hello {name}!
It's Your birthday today so here is an extra 100€ to Your upcoming payment
```
How many things would need to change in order to do that?

But very good job, really nice and clean code!